log/context.go

```diff
@@ -37,5 +37,9 @@ func NamesFromContext(ctx context.Context) []string {
 }
 
 func with(ctx context.Context, lvl Level, names ...string) context.Context {
+	if ctx == nil {
```

Comment:
Why do you need the check? Allowing a nil context usually leads to a panic if someone passes it. The standard library and others typically don't add guards for this situation. This helps prevent mistakes when someone forgets to pass a context, which can be hard to debug.